array_test.go

```diff
@@ -842,234 +842,237 @@ func TestArray_InList(t *testing.T) {
-		value.chain.clear()
-
-		value.NotConsistsOf("foo", 123)
-		value.chain.assert(t, success)
-		value.chain.clear()
-
-		value.ConsistsOf(123, "foo", "foo")
-		value.chain.assert(t, failure)
-		value.chain.clear()
-
-		value.NotConsistsOf(123, "foo", "foo")
-		value.chain.assert(t, success)
-		value.chain.clear()
-
-		value.ConsistsOf(123, "foo")
-		value.chain.assert(t, success)
-		value.chain.clear()
-
-		value.NotConsistsOf(123, "foo")
-		value.chain.assert(t, failure)
-		value.chain.clear()
 
+		cases := []struct {
+			name           string
+			value          []interface{}
+			wantConsistsOf chainResult
+		}{
+			{
+				name:           "first value (int) in array",
+				value:          []interface{}{123},
+				wantConsistsOf: failure,
+			},
+			{
+				name:           "second value (string) in array",
+				value:          []interface{}{"foo"},
+				wantConsistsOf: failure,
+			},
+			{
+				name:           "reversed order",
+				value:          []interface{}{"foo", 123},
+				wantConsistsOf: failure,
+			},
+			{
+				name:           "added element to test edge case",
+				value:          []interface{}{"foo", "foo", 123},
+				wantConsistsOf: failure,
+			},
+			{
+				name:           "copy of array",
+				value:          []interface{}{123, "foo"},
+				wantConsistsOf: success,
+			},
+		}
+		for _, tc := range cases {
+			t.Run(tc.name, func(t *testing.T) {
+				reporter := newMockReporter(t)
+
+				NewArray(reporter, []interface{}{123, "foo"}).ConsistsOf(tc.value...).
+					chain.assert(t, tc.wantConsistsOf)
+				NewArray(reporter, []interface{}{123, "foo"}).NotConsistsOf(tc.value...).
+					chain.assert(t, !tc.wantConsistsOf)
+			})
+		}
 	})
 
 	t.Run("canonization", func(t *testing.T) {
 		type (
 			myInt int
 		)
-
 		reporter := newMockReporter(t)
 
-		value := NewArray(reporter, []interface{}{123, 456})
+		assert.Equal(t, []interface{}{123.0, 456.0},
+			NewArray(reporter, []interface{}{123, 456}).Raw())
 
-		assert.Equal(t, []interface{}{123.0, 456.0}, value.Raw())
+		NewArray(reporter, []interface{}{123, 456}).ConsistsOf(myInt(123), 456.0).
+			chain.assert(t, success)
 
-		value.ConsistsOf(myInt(123), 456.0)
-		value.chain.assert(t, success)
-		value.chain.clear()
-
-		value.NotConsistsOf(myInt(123), 456.0)
-		value.chain.assert(t, failure)
-		value.chain.clear()
+		NewArray(reporter, []interface{}{123, 456}).NotConsistsOf(myInt(123), 456.0).
+			chain.assert(t, failure)
 	})
 
 	t.Run("invalid argument", func(t *testing.T) {
 		reporter := newMockReporter(t)
 
-		value := NewArray(reporter, []interface{}{})
-
-		value.ConsistsOf(func() {})
-		value.chain.assert(t, failure)
-		value.chain.clear()
+		NewArray(reporter, []interface{}{}).ConsistsOf(func() {}).
+			chain.assert(t, failure)
 
-		value.NotConsistsOf(func() {})
-		value.chain.assert(t, failure)
-		value.chain.clear()
+		NewArray(reporter, []interface{}{}).NotConsistsOf(func() {}).
+			chain.assert(t, failure)
 	})
 }
 
 func TestArray_Contains(t *testing.T) {
```

Comment:
Sorry, I completely forgot about one thing when I was creating this issue.

In next major release (v3), Array.(Not)Contains will be changed to accept only a single argument. Multiple arguments will be allowed only for ContainsAny, ContainsAll, etc.

Since you're modifying these tests now, could you please change the tests to check only for single argument cases? So that the tests will be ready for v3 change.